docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+parameter names become unique. E.g. no need for cryptic `par_00X_` parameter names
+in the minuit backend, which simplifies debugging and interaction by the user with
+methods such as `Fit.likelihood_profile()` or `Fit.confidence()`, where parameter
+identifiers must be given.
+
+
+Improve and implement model I/O
+-------------------------------
+Serialization of models is an important feature for standard analyses. In
+`gammapy.utils.serialization` there is a prototype implementation of XML
+serialization for `SourceModels`, We propose to fix this existing XML I/O
+prototype so that the following easily works:
+
+.. code-block::
+
+    from gammapy.cube import SourceModels
+
+    model = SourceModels.read("model.xml")
+    print(model)
+
+    model.write("model.xml")
+
+
+In addition we propose to implement an additional YAML serialization format, which
+results in human-readable model configuration files. Once available the YAML format
+should be the preferred serialization:
+
+.. code-block::
+
+    model = SourceModels.read("model.yaml")
+    print(model)
+
+    model.write("model.yaml")
+
+The two file formats should be handled automatically in `SourceModel.read()` and
+`SourceModel.write()`.
+
+
+Improve spatial models
+----------------------
+
+Implement sky coordinate handling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Implement support for coordinate systems. Add an interface for `SkyCoord`, by
+introducing a `SpatialModel.skycood` attribute.
+
+.. code-block::
+
+    from astropy.coordinates import SkyCoord
+    from gammapy.image.models import PointSource
+
+    # instanciation using a SkyCoord object
+    point_source = PointSource(skycood=SkyCoord(0, 0, frame="galactic", unit="deg"))
+    # or
+    point_source = PointSource(lon="0 deg", lat="0 deg", frame="galactic")
+
+    skycoord = point_source.skycoord
+    assert isinstance(skycoord, SkyCoord)
+
+    # evaluation of models using sky coordinates
+    coords = SkyCoord([0, 1], [0.5, 0.5], frame="galactic", unit="deg")
+    values = point_source(coords)
+
+    # maybe override evaluation for efficient evaluation for fitting
+    values = point_source(lon, lat)
+
+
+Implement default parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For spectral models Gammapy has defined default parameters, which simplifes the
+interactive usage of the models. From this experience we propose to introduce
+default parameters for `SpatialModel` as well as `SourceModel`.
+
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+    from gammapy.cube.models import SourceModels
+
+    shell = Shell()
+    print(shell)
+
+
+Implement evaluation region specifications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For efficient evaluation of model components, we propose that `SpatialModel`
+classes declare a circular region `.evaluation_radius`, they are evaluated on.
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+
+    shell = Shell()
+    print(shell.evaluation_radius)
```

Comment:
Remove `print` and isntead state in the text want it is. Presumably an `astropy.coordinates.Angle`, no?